# Requirements and dataset preparation

---

activate the environiment: stasis.yaml

A little bit of coding is required to reformat the input files

---

Eulimnadia texana genome was available [here](https://wfitch.bio.uci.edu/~tdlong/PapersRawData/BaldwinShrimpAssembly.tar.gz).
The .gff files for
Brachinecta [GCA_023053575.1](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/023/053/575/GCA_023053575.1_BRLY_1.1/GCA_023053575.1_BRLY_1.1_genomic.fna.gz) 
and 
Chidrorus [GCA_030141595.1](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/030/141/595/GCA_030141595.1_ASM3014159v1/GCA_030141595.1_ASM3014159v1_genomic.fna.gz) 
spp. were received directly from corresponding authors of [this]() and [this]() paper. 

For these two species the gff and fna were not exactly matching, soooo

`awk -F " " '/^>/{print ">"$6;next;} {print;}' GCA_023053575.1_BRLY_1.1_genomic.fna | tr -d "," > GCA_023053575.1_BRLY_1.1_genomic.ref.fna`

Then the longest isoform for each gene was extracted using [agat]()

`agat_sp_keep_longest_isoform.pl -gff sp.gtf -o genomic_longest_isoform.gff`

`agat_sp_extract_sequences.pl -g genomic_longest_isoform.gff -f sp.fna -t cds -o sp_longest_isoform_cds.fa`

---

The other genomes were downloaded from NCBI using [database]() and then isoforms were collapsed using [agat]()

`sh download_genomes.sh refseq_genomes.tsv`

`for i in $(ls ncbi_dataset/data/*/*gff); do agat_sp_keep_longest_isoform.pl -gff $i -o ${i::-4}_longest_isoform.gff; done`

`while read line; do acc=$(echo $line | awk '{print $1}'); spp=$(echo $line | awk '{print $2}'); agat_sp_extract_sequences.pl -g ncbi_dataset/data/$acc/genomic_longest_isoform.gff -f ncbi_dataset/data/$acc/*.fna -t cds -o longest_isoform_cds/$sp_longest_isoform_cds.fa; done < genomes.tsv`

--- 

We checked genomes completeness using BUSCO

For Dsin we had to remove pipes from fasta headers to have it work with busco, using `sed -i 's/|/_/g' Dsin_longest_isoform_cds.fa`

Then busco was run using:

`for i in *fa; do busco -i $i -o ${i::4}.busco -m genome --auto-lineage-euk; done`

---

Then we moved translated all CDSs using transeq:

`for i in *fa; do busco -i $i -o ${i::4}.busco -m genome --auto-lineage-euk; done`

---

here is a recap of all the species included in analyses:


lapu	Lepidurus apus			Notostraca
lart	Lepidurus articus		Notostraca
lcou	Lepidurus couesii		Notostraca
lubb	Lepidurus lubbockii		Notostraca
tgra	Triops granarius		Notostraca
tita	Triops cancriformis		Notostraca
tlon	Triops longicaudatus		Notostraca
tnum	Triops numidicus		Notostraca
tusa	Triops cancriformis		Notostraca
ldal	Lepthesteria dahalacensis	Spinicaudata
Etex    Eulimnadia texana               Spinicaudata
Dcar	Daphnia carinata		Anomophoda
Dgal	Daphnia galeata			Anomophoda
Dmag	Daphnia magna			Anomophoda
Dpua	Daphnia pulicaria		Anomophoda
Dpux	Daphnia pulex			Anomophoda
Dsin	Daphnia sinensis		Anomophoda
Csph	Chydorus sphaericus		Anomophoda
Blin	Branchinecta lynchi		Anostraca
Afra	Artemia franciscana		Anostraca

4 Lepidurus 
5 Triops 
6 Daphnia
2 Spinicaudata
2 Anostraca
