### 01 orthology inference
```OrthoFinder_source/orthofinder.py -S diamond_ultra_sens -y -f ortho/```

### 02 make one-liner the fasta files with more than 4 tips
```for i in $(ls Orthogroup_Sequences/*); do n=$(grep -c ">" $i); if [[ $n -ge 4 ]]; then awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < $i | awk -F "_" '{print $1}' > reformat/"${i: 21}"; fi; done```

### 03 for functional annotation - generate a fasta with one sequence for each orthogroup 
```for i in *ref.fa; do sed "s/^>/&"${i%.ref.fa}"_/g" $i | tail -2; done > functional_annotation/AA4GO.fa```

### 04 reformat OMA outpout
```for i in $(awk -F '_' '{print $2}' ff705ee4099ad63bdb7ea0d9426b29a7.txt | awk -F " " '{print $2}' | uniq); do GO=$(grep $i ff705ee4099ad63bdb7ea0d9426b29a7.txt | awk -F " " '{print $4}' | sed -z '$ s/\n$//g' | sort -u | sed "s/  /,/g "); echo -e $i"\t"$GO | sed 's/ /,/g'; done > GO_Universe```

### 05 align the reformatted fasta
```for i in $(ls reformat/*ref.fa); do mafft $i > ${i::-2}aln.fa; done```

### 06 clean alignements
```for i in $(ls reformat/*aln.fa); do trimal -gappyout -resoverlap 0.5 -seqoverlap 50 -in $i -out ${i::-6}trm.fa; done```

### 07 infer a gene tree
```for i in $(ls *trm.fa); do iqtree2 -s $i -mset LG,WAG,JTT -nt AUTO; done```

### 08 decompose gene trees
```for i in *treefile; do python disco.py -i $i -o ${i%treefile}.disco -d _ -n 1 -m 6 --keep-labels; done```

### 09 reconstruct decomposed msa
```n=0; for k in *disco; do while read line; do ((n++)); for j in $(echo $line | sed "s/:[0-9]*\.[0-9]*//g" | sed "s/e-[0-9]*//g" | sed 's/(//g' | sed 's/)/\n/g' | sed 's/,/\n/g' | sed "s/:[0-9]//g" | tr -d ";" | sort -u); do sed -n -e "/$j/,/>/ p" ${k::-6} | head -n -1 | awk -F "_" '{print $1}' | sed 's/-//g' | awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' >> "${k::-17}"_"$n".disco.fa; done; done < $k; n=0; done```

### 10 align decomposed msa
```for i in $(ls *disco.fa); do mafft $i > ${i::-2}aln.fa; done```

### 11 clean decomposed msa
```for i in $(ls *disco.aln.fa); do trimal -gappyout -resoverlap 0.5 -seqoverlap 50 -in $i -out ${i::-6}trm.fa; done```

### 12 infer TRACCER input
```for i in $(ls *disco.trm.fa); do grep ">" $i | tr -d ">" > tmp.msa.sp; grep -v -f tmp.msa.sp sp.lst > tmp.prune; gotree prune -i species_tree.treefile -f tmp.prune > tmp.pruned.sptree; iqtree -mset LG,JTT,WAG -s $i -te tmp.pruned.sptree -keep-ident -safe; done; rm *tmp```

### 11 put all DISCO trees in a single file 
```for i in *.disco.trm.fa.treefile; do while read line; do echo -e ${i::-22} "\t" $line; done < $i; done > TRACCER_input.nwk```

### 12 run convergent rates analysis
```python TRACCER.py --mastertree=species_tree.treefile --gtrees=TRACCER_input.nwk --hastrait=Lpac,lapu,lart,lcou,lubb,tgra,tita,tlon,tnum,tusa --outname=test --outgroup=Afra --min_species=6 --cpus=4 --min_Tspecies=3 --min_NTspecies=3```
